src/core/qgstiles.cpp

```diff
@@ -25,6 +27,7 @@ QgsTileMatrix QgsTileMatrix::fromWebMercator( int zoomLevel )
   double s0 = 559082264.0287178;   // scale denominator at zoom level 0 of GoogleCRS84Quad
 
   QgsTileMatrix tm;
+  tm.mCrs = QgsCoordinateReferenceSystem( "EPSG:3857" );
```

Comment:
It would be nice if also the the web mercator definition could be defined through the newly introduced fromCustomDef(), so it is not some kind special case. Just wondering if the calculated scale denominator would be the same (I don't know how exactly the scale denominator 559082264.0287178 was calculated for web mercator)...